iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -92,379 +77,376 @@ TEST_F(ServiceDiscovery_test, FindServiceWithWildcardsReturnsOnlyIntrospectionSe
+    auto serviceContainer = sut.findService(SERVICE, SERVICE_DESCRIPTION1.getInstanceIDString());
     ASSERT_FALSE(serviceContainer.has_error());
-    ASSERT_THAT(serviceContainer.value().size(), Eq(1u));
-    ASSERT_THAT(*serviceContainer.value().begin(), Eq(ServiceDescription{"service2", "instance1", "event1"}));
+    ASSERT_THAT(serviceContainer.value().size(), Eq(1U));
+    EXPECT_THAT(*serviceContainer.value().begin(), Eq(SERVICE_DESCRIPTION1));
 
-    serviceContainer = sut.findService(IdString_t("service2"), IdString_t("instance2"));
+    serviceContainer = sut.findService(SERVICE, SERVICE_DESCRIPTION2.getInstanceIDString());
     ASSERT_FALSE(serviceContainer.has_error());
-    ASSERT_THAT(serviceContainer.value().size(), Eq(1u));
-    ASSERT_THAT(*serviceContainer.value().begin(), Eq(ServiceDescription{"service2", "instance2", "event2"}));
+    ASSERT_THAT(serviceContainer.value().size(), Eq(1U));
+    EXPECT_THAT(*serviceContainer.value().begin(), Eq(SERVICE_DESCRIPTION2));
 
-    serviceContainer = sut.findService(IdString_t("service2"), IdString_t("instance3"));
+    serviceContainer = sut.findService(SERVICE, SERVICE_DESCRIPTION3.getInstanceIDString());
     ASSERT_FALSE(serviceContainer.has_error());
-    ASSERT_THAT(serviceContainer.value().size(), Eq(1u));
-    ASSERT_THAT(*serviceContainer.value().begin(), Eq(ServiceDescription{"service2", "instance3", "event3"}));
+    ASSERT_THAT(serviceContainer.value().size(), Eq(1U));
+    EXPECT_THAT(*serviceContainer.value().begin(), Eq(SERVICE_DESCRIPTION3));
 }
 
-TEST_F(ServiceDiscovery_test, StopOfferWithInvalidServiceDescriptionFails)
+TEST_F(ServiceDiscovery_test, FindServiceReturnsCorrectServiceInstanceCombinations)
 {
-    ::testing::Test::RecordProperty("TEST_ID", "7f758831-674b-4ea2-b5ee-1be0b22d8292");
-    EXPECT_FALSE(
-        sut.stopOfferService({iox::capro::InvalidIdString, iox::capro::InvalidIdString, iox::capro::InvalidIdString}));
-}
+    ::testing::Test::RecordProperty("TEST_ID", "360839a7-9309-4e7e-8e89-892097a87f7a");
+
+    const IdString_t SERVICE1 = "service1";
+    const IdString_t SERVICE2 = "service2";
+    const IdString_t INSTANCE1 = "instance1";
+    const IdString_t INSTANCE2 = "instance2";
+    const IdString_t INSTANCE3 = "instance3";
+    const IdString_t EVENT1 = "event1";
+    const IdString_t EVENT2 = "event2";
+    const IdString_t EVENT3 = "event3";
+
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION1(SERVICE1, INSTANCE1, EVENT1);
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION2(SERVICE1, INSTANCE2, EVENT2);
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION3(SERVICE1, INSTANCE3, EVENT3);
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION4(SERVICE2, INSTANCE1, EVENT1);
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION5(SERVICE2, INSTANCE2, EVENT2);
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION6(SERVICE2, INSTANCE3, EVENT3);
```

Comment:
What do you think about using `SERVICE_A` etc., to not let `SERVICE_DESCRIPTION2` have `SERVICE1`?

Or maybe even better having `SERVICE_DESCRIPTION_{service ID}_{instance ID}`. Then it's easier to understand the matches